### PR TITLE
Move to Common/AI/ and forward trait notifications.

### PR DESCRIPTION
**Pull request by Phrohdoh**
Also change the attack module to use attack-move, this doesn't help if an actor dies or becomes invalid.

### CODE REVIEW

**Comment by Unit158** [comment]
Code looks reasonable and it works. I'm not a fan of the base class knowing the main base, as I can think of cases where it shouldn't, but that is unrelated.

File: OpenRA.Mods.Common/AI/ModularAI/IdleBaseBuilderAIModule.cs [hunk 1]
```diff
@@ -57,15 +54,14 @@ public void Tick(Actor self)
...
+				var transforms = builder.Trait<Transforms>();
+				var deploysInto = transforms.Info.IntoActor;
 
 				if (mainBaseBuilding != null)
 				{
 					if (builder.IsMoving())
 						continue;
 
-					var transforms = builder.Trait<Transforms>();
-					var deployInto = transforms.Info.IntoActor;
```
Comment (penev92) [inline OpenRA.Mods.Common/AI/ModularAI/IdleBaseBuilderAIModule.cs]: This move outside the `if` seems redundant.

**Comment by penev92** [comment]
Otherwise looks good :+1:

File: OpenRA.Mods.Common/AI/ModularAI/IdleBaseBuilderAIModule.cs (hunk 1 above)
Comment (Phrohdoh) [inline OpenRA.Mods.Common/AI/ModularAI/IdleBaseBuilderAIModule.cs]: Oh right, it was for a test and I forgot to revert it.

**Comment by Phrohdoh** [comment]
Updated.

**Pull request merged**